gradio/blocks.py

```diff
@@ -429,6 +429,8 @@ def postprocess_update_dict(
         else:
             update_dict["value"] = value
     update_dict["__type__"] = "update"
+    if "root" in update_dict:  # reserved keyword
+        update_dict["_root"] = update_dict["root"]
```

Comment:
As discussed in retro, I think a better approach would be to:

1. soft deprecate the `root` parameter with a warning
2. add a `root_dir` parameter in `FileExplorer` (this also matches the name of the parameter in `glob.glob()` so that's a nice plus)
3. in the constructor of `FileExplorer`, check to see if `root_dir` is not supplied and if `root` is. If that's the case, do `self.root_dir = root`
4. in the frontend, replace `root` with `root_dir`

I think in that case, you shouldn't have to make any hacky changes here

Reply:
done